blobfuse/fileapis.cpp

```diff
@@ -335,21 +335,24 @@ int azs_flush(const char *path, struct fuse_file_info *fi)
             }
       
             errno = 0;
-            std::vector<std::pair<std::string, std::string>> metadata;
-            storage_client->UpdateBlobProperty(blob_name, "", "", &metadata);
-            storage_client->UploadFromFile(mntPath, metadata);
-            if (errno != 0)
-            {
-                int storage_errno = errno;
-                syslog(LOG_ERR, "Failing blob upload in azs_flush with input path %s because of an error from upload_file_to_blob().  Errno = %d.\n", path, storage_errno);
-                free(path_buffer);
-                return 0 - map_errno(storage_errno);
-            }
-            else
-            {
-                syslog(LOG_INFO, "Successfully uploaded file %s to blob %s.\n", path, blob_name.c_str());
+            if (buf.st_mtime > ((struct fhwrapper *)fi->fh)->mtime_at_upload) {
```

Comment:
I haven't tested this yet, so expect comments a little later.

Reply:
This check is introduced to solve the double upload issue. When we upload we will store the last modified time of file and when the next upload request comes we just check anything in file has changed or not, if no then skip the upload part. As soon as file handle is closed this data will be wiped out so this will hit only when same file is flushed twice before close.